LoopKit/WeakObserverSet.swift

```diff
@@ -0,0 +1,147 @@
+//
+//  WeakObserverSet.swift
+//  LoopKit
+//
+//  Created by Michael Pangburn
+//  Copyright © 2019 LoopKit Authors. All rights reserved.
+//
+
+import Foundation
+
+
+/// A set-like collection of weak types, providing closure-baesd iteration on a client-specified queue
+/// Mutations and iterations are thread-safe
+public class WeakObserverSet<Observer> {
+    private typealias Identifier = ObjectIdentifier
+    private typealias Element = ObserverDispatchContainer<Observer>
+
+    private class ObserverDispatchContainer<Observer> {
+        private weak var _observer: AnyObject?
+        weak var queue: DispatchQueue?
```

Comment:
Why is the queue reference `weak`?

Reply:
A queue is likely either owned by the observer or it’s global, and in the former we don’t want to continue to hang on to the reference.